### PR TITLE
Plugins: Refactor env vars to dedicated service

### PR DESCRIPTION
**Pull request by wbrowne**
**What is this feature?**

Refactors plugin env vars into a dedicated service. 

Also exposes a new signature for creating unmanaged backend plugins (this can be in a separate PR if necessary)

**Why do we need this feature?**

To be able to override default behaviour of the default backend plugin provider (Grafana managed Hashicorp process)

Please check that:
- [X] It works as expected from a user's perspective.
- [X] If this is a pre-GA feature, it is behind a feature toggle.
- [X] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -121,7 +121,7 @@ require (
 	gopkg.in/mail.v2 v2.3.1
 	gopkg.in/yaml.v2 v2.4.0 // indirect
 	gopkg.in/yaml.v3 v3.0.1
-	xorm.io/builder v0.3.6 // indirect
+	xorm.io/builder v0.3.6
```
Comment (marefr) [inline go.mod:124]: Nit. Why did this change? Does running `go mod tidy` make this go away?

**Review by marefr — APPROVED** [review]
LGTM added a comment

File: go.mod (hunk 1 above)
Comment (wbrowne) [reply go.mod:124]: Looks like `main` needs a `go mod tidy` 🤔

**Review by andresmgot — APPROVED** [review]
LGTM

**Pull request merged**